packages/middleware/src/index.ts

```diff
@@ -37,25 +49,26 @@ async function getMiddlewareFile() {
   return middlewareFiles[0];
 }
 
-export async function build() {
-  const middlewareFile = await getMiddlewareFile();
+export async function build({ workPath }: BuildOptions) {
```

Comment:
I was trying to avoid having to specify `workPath` explicitly since it should be able to be inferred from cwd. But let's leave it since it'll make writing tests easier (makes the function more "pure").

Using `BuildOptions` interface though isn't quite correct, since that's the interface for the Runtimes' `build()` function, not the Plugins' `build()` function. So far, this `workPath` is the only prop that needs to be passed in to the Plugin build fn.